consumer/processing.go

```diff
@@ -30,13 +30,19 @@ import (
 // Report represents report send in a message consumed from any broker
 type Report map[string]*json.RawMessage
 
+// MaybeRequestID contains either a string or nil depending on the content
+// of the input message. Sometimes the request ID is missing,
+// in which case it is nil, otherwise it should always be a string.
+type MaybeRequestID interface{}
```

Comment:
What about to use a pointer to a string or a struct with field `IsNull`?

Reply:
In the end, it runs out that using a regular `string` works just fine.